### ISSUE TITLE
Animation on expand and shrink

### ISSUE THREAD

**Issue by naamapps** [fweinaug/expandable_text#14]
Hey nice package!
It will be great if you could add an animation (change height and fade?) when expanding and shrinking the text.
That will make the UX better and smoother on the eyes.

Thanks :)

**Comment by fweinaug** [comment]
Thanks! I just released version 2.1 which animates expanding and collapsing the text. It has to be enabled with the `animation` flag. Let me know if everything works for you.

**Comment by naamapps** [comment]
Wow!
Really impressive @fweinaug.
Just nitpicking here, but it will be best to add a curve property for the animation.
And is it possible to add a fade effect to the animation?
Right now the expanded text shows immediately and it is a bit jarring.

The height animation is perfect.

I really appreciate your work.
Thanks

**Comment by fweinaug** [comment]
> Just nitpicking here, but it will be best to add a curve property for the animation.
Good point, I'll add it with the next update.

> And is it possible to add a fade effect to the animation?
> Right now the expanded text shows immediately and it is a bit jarring.
Do you mean animate the height and also fade in/out the text? Do you mean a fade through/crossfade animation?

**Comment by naamapps** [comment]
@fweinaug Yes animate the height + fade in/out (fade in only the new visible text if expanding and when collapsing only fade out the text that is being cut out).
see this package: https://pub.dev/packages/expandable
They fade in/out the overflowed text.
Hope it's more understandable now.
Thanks again
